src/fsharp/ExtensionTyping.fs

```diff
@@ -246,11 +246,12 @@ module internal ExtensionTyping =
     // providers can implement wrap-and-filter "views" over existing System.Type clusters without needing
     // to preserve object identity when presenting the types to the F# compiler.
 
-    let providedSystemTypeComparer = 
-        let key (ty: System.Type) = (ty.Assembly.FullName, ty.FullName)
-        { new IEqualityComparer<Type> with 
-            member __.GetHashCode(ty: Type) = hash (key ty)
-            member __.Equals(ty1: Type, ty2: Type) = (key ty1 = key ty2) }
+    type ProvidedTypeComparer() = 
```

Comment:
I'm not convinced making this a type rather than a let binding is advantageous, making it a type causes us to need the .Instance.  and is the reason all of the subsequent ands were needed.

Reply:
> I'm not convinced making this a type rather than a let binding is advantageous, making it a type causes us to need the .Instance. and is the reason all of the subsequent ands were needed.

in the ProvidedTypeComparer code, we refer to the Provided Type, which declared below. Because of this, I made this a type and added "and"